packages/server/src/lib/access-helpers.js

```diff
@@ -128,6 +129,7 @@ async function isMicrosoftSafeLinksRequest(req, res, next) {
     const userAgent = req.headers['user-agent'] || '';
     const nativeHost = req.headers['x-native-host'] || '';
     if (userAgent.toLowerCase().includes('oneoutlook') || nativeHost.toLowerCase().includes('oneoutlook')) {
+        log.info({ headers: req.headers }, 'Microsoft Safe Links request');
```

Comment:
Overabundance of caution, but because `req.headers` _might_ contain values that we'd want to remain out of logs, I'd recommend constraining this a bit so that only the `user-agent` header value is logged:
```suggestion
        log.info({ 'user-agent': req.headers['user-agent'] }, 'Microsoft Safe Links request');
```

or, for a bit more insight into the complete set of headers:
```suggestion
        log.info({ 'user-agent': req.headers['user-agent'], headers: Object.keys(req.headers) }, 'Microsoft Safe Links request');
```

Reply:
Good idea 👍  I took your suggestion and added an additional key `nativeHost`

Reply:
Ah, good idea adding that. Will approve!